saul-core/src/main/scala/edu/illinois/cs/cogcomp/saul/datamodel/node/Node.scala

```diff
@@ -275,7 +282,7 @@ class Node[T <: AnyRef](val keyFunc: T => Any = (x: T) => x, val tag: ClassTag[T
   }
 
   def loadDerivedInstances(in: ExceptionlessInputStream) = {
-    count = in.readInt()
+    count = new AtomicInteger(in.readInt())
```

Comment:
Something to remember for future: replacing may of our integers/counters with `AtomicInteger` might be a good idea.